attestation/context.go

```diff
@@ -205,48 +165,34 @@ func (ctx *AttestationContext) runAttestor(attestor Attestor) error {
 		ctx.addMaterials(materialer)
 	}
 
-	if producter, ok := attestor.(Producer); ok {
-		ctx.addProducts(producter)
+	if producer, ok := attestor.(Producer); ok {
+		ctx.addProducts(producer)
 	}
 
 	return nil
 }
 
 func (ctx *AttestationContext) CompletedAttestors() []CompletedAttestor {
-	attestors := make([]CompletedAttestor, len(ctx.completedAttestors))
-	copy(attestors, ctx.completedAttestors)
-	return attestors
+	return ctx.completedAttestors
 }
 
 func (ctx *AttestationContext) WorkingDir() string {
 	return ctx.workingDir
 }
 
 func (ctx *AttestationContext) Hashes() []crypto.Hash {
-	hashes := make([]crypto.Hash, len(ctx.hashes))
-	copy(hashes, ctx.hashes)
-	return hashes
+	return ctx.hashes
 }
 
 func (ctx *AttestationContext) Context() context.Context {
 	return ctx.ctx
 }
 
 func (ctx *AttestationContext) Materials() map[string]cryptoutil.DigestSet {
-	matCopy := make(map[string]cryptoutil.DigestSet)
-	for k, v := range ctx.materials {
-		matCopy[k] = v
-	}
-
-	return matCopy
+	return ctx.materials
```

Comment:
In contract to me overthinking it when it came to slices, it turns out I wasn't over thinking it when it came to maps. By returning the map, users can modify the unexported map within the struct by modifying the return value of this function:

https://go.dev/play/p/Ga5WdxpE9s4

In this case, we probably still want to copy the map before returning it.